### PR TITLE
Support folder upload

### PR DESCRIPTION
**Pull request by geoidesic**
These changes support the ability to upload a folder structure containing multiple files and to preserve that folder structure. The changes are backwards compatible, so that normal single or multiple file uploads work as before.

### CODE REVIEW

File: filer/static/filer/js/addons/table-dropzone.js [hunk 1]
```diff
@@ -80,6 +87,39 @@
...
+                        }
+                        return file.name
+                    },
+                    // ensure that the path informatio is sent as part of the request, 
+                    // without this the path is stripped out automatically by Djano.
+                    params(files, xhr, chunk) {
+                        var metadata = {}
+                        files.forEach(function(file, index) {
+                            let fullPath = file.upload.filename
+                            let path = fullPath.substr(0, fullPath.lastIndexOf('/'))
```
[the remark this answers is not in the document]
Comment (chaosk) [reply filer/static/filer/js/addons/table-dropzone.js]: Also it's not entirely clear that you're removing filename from the path - can you add a comment?

[the remark this answers is not in the document]
Comment (geoidesic) [reply filer/static/filer/js/addons/table-dropzone.js]: all done

File: filer/static/filer/js/addons/table-dropzone.js [hunk 2]
```diff
@@ -79,7 +88,33 @@
                     previewTemplate: '<div></div>',
                     clickable: false,
                     addRemoveLinks: false,
-                    parallelUploads: dropzone.data(dataUploaderConnections) || 3,
+                    parallelUploads: dropzone.data(dataUploaderConnections) || 1,
+                    // ensure that the path informatio is sent as part of the request,
```
Comment (chaosk) [inline filer/static/filer/js/addons/table-dropzone.js]: typos here

Comment (geoidesic) [reply filer/static/filer/js/addons/table-dropzone.js]: fixed

File: filer/static/filer/js/libs/dropzone.min.js [hunk 3]
```diff
@@ -1,86 +1,31 @@
...
...message($| )/.test(l.className)){t=l,l.className="dz-message";break}t||(t=Dropzone.createElement("<div class=\"dz-message\"><span></span></div>"),this.element.appendChild(t));let a=t.getElementsByTagName("span")[0];return a&&(null==a.textContent?null!=a.innerText&&(a.innerText=this.options.dictFallbackMessage):a.textContent=this.options.dictFallbackMessage),this.element.appendChild(this.getFallbackForm())},resize(t,a,l,s){let d={srcX:0,srcY:0,srcWidth:t.width,srcHeight:t.height},p=t.width/t.height;null==a&&null==l?(a=d.srcWidth,l=d.srcHeight):null==a?a=l*p:null==l&&(l=a/p),a=Math.min(a,d.srcWidth),l=Math.min(l,d.srcHeight);let u=a/l;if(d.srcWidth>a||d.srcHeight>l)if("crop"===s)p>u?(d.srcHeight=t.height,d.srcWidth=d.srcHeight*u):(d.srcWidth=t.width,d.srcHeight=d.srcWidth/u);else if("contain"===s)p>u?l=a/p:a=l*p;else throw new Error(`Unknown resizeMethod '${s}'`);return d.srcX=(t.width-d.srcWidth)/2,d.srcY=(t.height-d.srcHeight)/2,d.trgWidth=a,d.trgHeight=l,d},transformFile(t,a){return(this.options.resizeWidth||this.options.resizeHeight)&&t.type.match(/image.*/)?this.resizeImage(t,this.options.resizeWidth,this.options.resizeHeight,this.options.resizeMethod,a):a(t)},previewTemplate:`\
```
Comment (vxsx) [inline filer/static/filer/js/libs/dropzone.min.js]: this is es6 so it won't work in slightly older browsers, also there's no mention about the version of the dropzone.js anymore which should still be there

**Pull request closed without merging**